src/Powercord/plugins/pc-spotify/Modal/Modal.jsx

```diff
@@ -89,11 +89,12 @@ module.exports = class Modal extends React.Component {
 
   updateData (playerState) {
     if (playerState && playerState.currently_playing_type === 'track') {
-      return this.setState({
+      return powercord.pluginManager.get('pc-spotify').getSpotifyLogo().then(value => {
```

Comment:
I'm not a fan of fetching the logo everytime the player state is updated. What you should to is save that in an object property, and re-use it if the img is `null`

Reply:
yikes i don't get that my JS knowledge is failing me 2.0